src/allencell_ml_segmenter/prediction/service.py

```diff
@@ -17,19 +17,13 @@ class ModelFileService(Subscriber):
     Parses the chosen model file to extract the preprocessing method.
     """
 
-    def __init__(self, model: PredictionModel):
+    def __init__(self, model: FileInputModel):
```

Comment:
this should be a `PredictionModel` always, right? If so, I think it's better to be specific in the type annotation

Reply:
A good piece of evidence for composition over inheritance.